dev/integration_tests/display_cutout_rotation/test_driver/display_cutout_test_test.dart

```diff
@@ -0,0 +1,93 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: avoid_print
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:flutter_driver/flutter_driver.dart';
+
+// display_cutout needs a custom driver becuase cutout manipulations needs to be
+// done to a device/emulator in order for the tests to pass.
+Future<void> main() async {
+  if (!(Platform.isLinux || Platform.isMacOS)) {
+    // Not a fundemental limitation, developer shortcut.
+    print('This test must be run on a POSIX host. Skipping...');
+    exit(0);
+  }
+  final bool adbExists =
+      Process.runSync('which', <String>['adb']).exitCode == 0;
+  if (!adbExists) {
+    print(r'This test needs ADB to exist on the $PATH.');
+    exit(1);
```

Comment:
nit: Just a matter of good CLI hygiene, use `exitCode = 1; return`.

`exit` is something we should never be using, as it will completely terminate the program, skipping finalizers and cleanup. None of that occurs in this simple CLI app, but folks will copy/paste from other examples and there is no reason to encourage exit usage.

Reply:
I had no idea that this was good practice. Will change